src/Product.php

```diff
@@ -626,6 +629,26 @@ public function processImages($data)
         return $this;
     }
 
+    /**
+     * @return array|null
+     */
+    protected function getAvailableImage(array $data)
```

Comment:
Rename this to `getAvailableImages` since it actually returns multiple images. I also think it makes sense to add information to the docblock regarding that only images that are assigned to the current store will be returned.

Reply:
Please add `array` as return type (see another comment regarding the returning of `null`).

Reply:
It should return only one image, first one that is available in store, because then it takes `urlMiddle` property from the array. 
```
if (!$this->getField('image')) {
      $this->setField('image', $this->getFromArray($data, 'urlMiddle'));
}
```
There should be functionality to set multiple image urls?

Reply:
No actually it is fine like that. I just oversaw it.